nautobot/nautobot_data_validation_engine/tests/test_custom_validators.py

```diff
@@ -0,0 +1,460 @@
+"""
+Model test cases
+"""
+
+from django.contrib.contenttypes.models import ContentType
+from django.core.exceptions import ValidationError
+from django.test import TestCase
+
+from nautobot.dcim.models import Location, LocationType, Rack
+from nautobot.extras.models import Status
+from nautobot.extras.plugins.validators import wrap_model_clean_methods
+from nautobot.nautobot_data_validation_engine.models import (
+    MinMaxValidationRule,
+    RegularExpressionValidationRule,
+    RequiredValidationRule,
+    UniqueValidationRule,
+)
+
+
+class RegularExpressionValidationRuleModelTestCase(TestCase):
+    """
+    Test cases related to the RegularExpressionValidationRule model
+    """
+
+    def setUp(self) -> None:
+        wrap_model_clean_methods()
+        self.location_type = LocationType(name="Region")
+        self.location_type.validated_save()
+        self.location_type.content_types.set([ContentType.objects.get_for_model(Rack)])
+        return super().setUp()
+
+    def test_invalid_regex_matches_raise_validation_error(self):
+        RegularExpressionValidationRule.objects.create(
+            name="Regex rule 1",
+            content_type=ContentType.objects.get_for_model(Location),
+            field="name",
+            regular_expression="^ABC$",
+        )
+
+        location = Location(
+            name="does not match the regex",
+            location_type=LocationType.objects.get_by_natural_key("Region"),
+            status=Status.objects.get_by_natural_key("Active"),
+        )
+
+        with self.assertRaises(ValidationError):
```

Comment:
All of these should be updated to use `assertRaisesRegex` and verify that the *right* ValidationError is raised.